### ISSUE TITLE
 Error: "name" and "value" are required for setHeader().

### ISSUE THREAD

**Issue by benoitguigal** [resin-io/resin-sdk#97]
Here is my REPL session: 
```
> var resin = require('resin-sdk')
> var credentials = {'username': 'myusername', 'password': 'mypassword'}
> resin.auth.login(credentials) 
{ _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _progressHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined,
  _settledValue: undefined }
> Error: "name" and "value" are required for setHeader().
    at ClientRequest.OutgoingMessage.setHeader (_http_outgoing.js:333:11)
    at new ClientRequest (_http_client.js:101:14)
    at Object.exports.request (http.js:49:10)
    at Object.exports.request (https.js:136:15)
    at Request.start (/Users/benoit/git/figure/web/node_modules/request/request.js:904:30)
    at Request.end (/Users/benoit/git/figure/web/node_modules/request/request.js:1635:10)
    at end (/Users/benoit/git/figure/web/node_modules/request/request.js:676:14)
    at Immediate._onImmediate (/Users/benoit/git/figure/web/node_modules/request/request.js:690:7)
    at processImmediate [as _immediateCallback] (timers.js:358:17)
```

**Comment by benoitguigal** [comment]
I am using node v0.12.0

**Comment by Page-** [comment]
My guess is that it tries to make use of your token, before you've actually manage to send the request to get one, @jviotti is the one to deal with this?  If you want a quick workaround in the meantime you should be able to use `loginWithToken` with the auth token you can find in your preferences page.

**Comment by benoitguigal** [comment]
Yes it working for loginWithToken, thanks !

**Comment by jviotti** [comment]
@benoitguigal Hey there, thanks for using the Resin SDK!

I'm sadly unable to reproduce your issue in OS X 10.10.4 and NodeJS v0.12.0. What operating system, and potentially what `resin-sdk` version are you running?

Given that the token attempts to self update in about 1 hour interval, do you repeatedly get the same issue? If so, can you check if you have a `$HOME/.resin/token` file?

**Comment by Page-** [comment]
@jviotti, did you wipe your token completely before testing?

**Comment by benoitguigal** [comment]
I am using resin-sdk@2.2.0  
```
> du -sh ~/.resin/token
du: /Users/benoit/.resin/token: No such file or directory
``` 
~/.resin directory was created though

**Comment by benoitguigal** [comment]
Ok I now have a ~/.resin/token file with a token equal to the one in the preferences page

**Comment by benoitguigal** [comment]
but I am still getting the error after  `resin.auth.logout() ` and trying to log in again with my credentials

**Comment by benoitguigal** [comment]
It is strange to see this module appearing in the stack trace  
`/Users/benoit/git/figure/web/node_modules/request/request.js`
I am using resin-sdk in a larger project that also depends on `request`. Can it be related to a namespace conflict or version problem ?

**Comment by jviotti** [comment]
@Page- I tried wiping my token and even try reproducing with a brand new account without success.

@benoitguigal I've discovered and fixed a slight issue in the auth module which might be related to this issue.

Can you try that fix by installing the appropriate branch with:

```sh
$ npm install --save resin-io/resin-sdk#fix/auth-body
```

**Comment by benoitguigal** [comment]
I installed the fix but get the same error

**Comment by jviotti** [comment]
@benoitguigal What version of `request` are you using in your app?

According to your stack trace, the issue comes from a `request` module in the top level `node_modules/`, instead of `node_modules/resin-request/node_modules/request`.

Can you also look at `/Users/benoit/git/figure/web/node_modules/request/request.js:904:30`?

**Comment by benoitguigal** [comment]
benoit ~/git/figure/web $ npm list request
figure@0.3.0 /Users/benoit/git/figure/web
├── request@2.53.0 
└─┬ resin-sdk@2.3.0
  └─┬ resin-token@2.4.0
    └── request@2.60.0  

Line 904 of `/Users/benoit/git/figure/web/node_modules/request/request.js` is 
`self.req = self.httpModule.request(reqOptions)`

**Comment by jviotti** [comment]
@benoitguigal Great, I'm able to reproduce the issue by locking to `request@2.53.0`. I'll send a fix soon!

**Comment by benoitguigal** [comment]
Cool thanks ! Looking forward :-)

**Comment by jviotti** [comment]
@benoitguigal I've released `2.3.1` which has the fix to your issue. Please feel free to open an issue if you have another problem!

**Comment by benoitguigal** [comment]
thanks a lot for the rapid support !
